R/tidy-upkeep.R

```diff
@@ -3,9 +3,43 @@
 #' @param year Approximate year when you last touched this package. If `NULL`,
 #'   the default, will give you a full set of actions to perform.
 use_tidy_upkeep_issue <- function(year = NULL) {
-  check_is_package("use_tidy_upkeep_issue()")
+  make_upkeep_issue(year = year, tidy = TRUE)
+}
+
+#' Open an issue in GitHub with a checklist of upkeep
+#' tasks
+#'
+#' @description This opens an issue in your package
+#'   repository with a checklist of tasks for regular
+#'   maintenance of your package. This is a fairly
+#'   opinionated list of tasks but we believe taking care
+#'   of them will generally make your package better,
+#'   easier to maintain, and more enjoyable for your
+#'   users. Some of the tasks are meant to be performed
+#'   only once (and once completed shouldn't show up in
+#'   subsequent lists), and some should be reviewed
+#'   periodically. The tidyverse team uses a similar
+#'   function [use_tidy_upkeep_issue()] for our annual
+#'   package Spring Cleaning.
+#'
+#' @param year Optional year you are performing the upkeep
```

Comment:
Maybe elaborate a bit more on the impact of `year`. In fact, I'm actually not entirely clear on how `year` is going to play out in the non-tidy version of this function.

Reply:
I was just thinking about this as well. It could default to the current year, and maybe put the year in the issue title, and not in the body? Or remove entirely... I don't think it's serving much of a purpose as is

Reply:
I don't think it makes much sense to treat year the way we currently do in the tidy upkeep issue. Yes, I think it makes sense to timestamp the issue in some sense, either in its title or the body, with the date and possible the usethis version.

Reply:
For the non-tidy function, I removed it from the body and put it in the issue title, defaulting to the current year. I like the idea about tagging it with the usethis version as well. For the tidy upkeep issue, the year is baked into the format quite heavily... though perhaps we should also default to the current year rather than all years as it is now? We also talked about using a custom field in the `DESCRIPTION` to tag the latest upkeep for tidyverse packages, is that something to look at now?

Reply:
Ok, year now always appears in upkeep issue title, defaults to current year

* for `use_tidy_upkeep_issue()`, if `year` is specified it will be used in the title and used to select the subset of checklist items included (as before). If `year` is `NULL`, the current year will be in the issue title, but all current and previous checklists will be included.
* for `use_upkeep_issue()`, `year` is only used in the title, as there are not multiple year-based checklists.

I added a footer to the issue, ala reprex: f7cce2298d8. To avoid spurious changes in snapshots, I mocked the `checklist_footer()` function in the tests... I'm not sure if I'm now abusing `local_mocked_bindings()`!